packages/react/src/modal/ModalContent.js

```diff
@@ -51,13 +50,6 @@ const ModalContent = forwardRef((inProps, ref) => {
     ...rest,
   };
 
-  useClickOutside(contentRef, (event) => {
-    // Close the modal when clicking outside the content
-    if (closeOnOutsideClick) {
-      (typeof onClose === 'function') && onClose(event);
-    }
-  });
-
```

Comment:
My comment:
Remove `useClickOutside` here because it cannot support closing in the correct stack order.